File: 153.js
// Time Complexity : O(logN)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

/**
Algorithm : Binary Search

Perform Binary search to find min element.
**/

/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let left = 0;
    let right = nums.length - 1;
    
    // the goal is to use `left` to point at the minimn number 
    
    while (left < right) {
        const mid = Math.floor((left + right) / 2);
        if (nums[mid] > nums[right]) {
          left = mid + 1;
        } else {
          right = mid;
        }
    }

    return nums[left];
};

File: 153.java
// Time Complexity : O(log n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : taken algorithm from youtube
//



// Your code here along with comments explaining your approach
//used modified binary search 

class Solution {
    public int findMin(int[] nums) {
    
        
         int start=0;
        int end=nums.length-1;
        
        while(start<end){
           int mid=((end-start)/2)+start;
            
            if(nums[mid]>nums[end]){
                
                start=mid+1;
            }else
                end=mid;
            }
        return nums[end];
        }
    }